### ISSUE TITLE
Crash (assertion failure) in Dafny.Translator.FuelSetting.GetFunctionFuel

### ISSUE THREAD

**Issue by 0xabu** [Microsoft/dafny#23]
The current master branch of Dafny crashes in an assertion failure when verifying the program below:

```
predicate {:opaque} ValidRegs(regs:map<int,int>)
{
    forall r:int :: 0 <= r < 10 ==> r in regs
}
 
predicate mypredicate(regs:map<int,int>)
    requires ValidRegs(regs)
 
lemma mylemma()
{
    var regs:map<int,int>;
    assume mypredicate(regs);
}
```

**Comment by parno** [comment]
Here's a simplified example:

```
predicate {:opaque} ValidRegs()
{
    forall i:int :: true
}
 
predicate mypredicate()
    requires ValidRegs()
 
lemma mylemma()
{
    assume mypredicate();
}
```

The problem is that this code ends up bumping up the fuel twice, once during split expression, which increases fuel while inlining the function call, and again when we discover the function contains a quantifier and exists in an assume context.  This brings us to a fuel setting of 3, which is unexpected.

**Comment by parno** [comment]
Closed by  4ff304d9b.
